File: 3475-minimum-operations-to-make-binary-array-elements-equal-to-one-i/minimum-operations-to-make-binary-array-elements-equal-to-one-i.java
class Solution {
    public int minOperations(int[] nums) {
        int result = 0;
        boolean allOne = true;

        for (int i = 0; i < nums.length - 2; i++) {
            if (nums[i] == 1)
                continue;

            nums[i] ^= 1;
            nums[i + 1] ^= 1;
            nums[i + 2] ^= 1;
            result++;
            allOne = false;
        }

        if (nums[nums.length - 2] == 0 || nums[nums.length - 1] == 0)
            return -1;

        if (allOne) return 0;

        return result > 0 ? result : -1;
    }
}

File: 3475-minimum-operations-to-make-binary-array-elements-equal-to-one-i/minimum-operations-to-make-binary-array-elements-equal-to-one-i.go
func minOperations(nums []int) int {
    result := 0;

    for i := 0; i < len(nums) - 2; i++ {
        if nums[i] != 0 {
            continue;
        }

        nums[i] = 1;
        nums[i+1] ^= 1;
        nums[i+2] ^= 1;

        result += 1;
    }

    if nums[len(nums) - 2] + nums[len(nums) - 1] != 2 {
        return -1;
    }

    return result;
}